### PR TITLE
Add base autogen code

### PR DESCRIPTION
**Pull request by edwardfeng-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/edwardfeng-db/mlflow/pull/10762?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10762/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10762
```

</p>
</details>

### Related Issues/PRs

 #xxx
https://github.com/mlflow/mlflow/pull/10474

### What changes are proposed in this pull request?

- Add basic graphql autogen code which will be able to successfully generate the schema
- Right now nothing has been onboarded, will do that as a follow-up PR. So effectively this PR is a no-op for existing functionality

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/protos/proto_to_graphql/autogeneration_utils.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from string_utils import camel_to_snake, snake_to_pascal
+import sys
+import re
+
+INDENT = "    "
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/autogeneration_utils.py:5]:
```suggestion
INDENT = " " * 4
```

File: mlflow/protos/proto_to_graphql/code_generator.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/code_generator.py:1]:
```suggestion
```

File: mlflow/protos/proto_to_graphql/code_generator.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    def __init__(self):
+        self.queries = set([])  # method_descriptor
+        self.mutations = set([])  # method_descriptor
+        self.inputs = []  # field_descriptor
+        self.types = []  # field_descriptor
+        self.enums = set([])  # enum_descriptor
+        self.outputs = set([])  # field_descriptor
+        self.method_names = set([])  # package_name_method_name
+        self.queries = set([])  # method_descriptor
+        self.mutations = set([])  # method_descriptor
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/code_generator.py:24]: /ai remove unnecessary empty lists

File: mlflow/protos/proto_to_graphql/code_generator.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    with open(SCHEMA_EXTENSION, 'w') as file:
+        file.write(schema)
+
+
+def main():
+    generate_code()
+
+
+if __name__ == '__main__':
+    main()
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/code_generator.py:48]:
```suggestion
    main()

```

File: mlflow/protos/proto_to_graphql/parsing_utils.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+
+def get_method_type(method_descriptor):
+    return method_descriptor.GetOptions().Extensions[databricks_pb2.rpc].endpoints[0].method
+
+
+"""
+Given a method descriptor, add information being referenced into the GenerateSchemaState.
+"""
+def process_method(method_descriptor, state):
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/parsing_utils.py:13]:
```suggestion
def process_method(method_descriptor, state):
    """
    Given a method descriptor, add information being referenced into the GenerateSchemaState.
    """
```

File: mlflow/protos/proto_to_graphql/parsing_utils.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            state.mutations.add(method_descriptor)
+        populate_message_types(method_descriptor.input_type, state, True, set([]))
+        populate_message_types(method_descriptor.output_type, state, False, set([]))
+
+
+"""
+Given a field descriptor, recursively walk through the referenced message types and add
+information being referenced into the GenerateSchemaState.
+"""
+def populate_message_types(field_descriptor, state, is_input, visited):
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/parsing_utils.py:37]:
```suggestion
def populate_message_types(field_descriptor, state, is_input, visited):
    """
    Given a field descriptor, recursively walk through the referenced message types and add
    information being referenced into the GenerateSchemaState.
    """
```

Comment (harupy) [reply mlflow/protos/proto_to_graphql/parsing_utils.py:37]: Can we also fix other docstrings?

Comment (edwardfeng-db) [reply mlflow/protos/proto_to_graphql/parsing_utils.py:37]: Sounds good, thanks!

File: mlflow/protos/proto_to_graphql/code_generator.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+class GenerateSchemaState:
+    def __init__(self):
+        self.queries = set([])  # method_descriptor
+        self.mutations = set([])  # method_descriptor
+        self.inputs = []  # field_descriptor
+        self.types = []  # field_descriptor
+        self.enums = set([])  # enum_descriptor
+        self.outputs = set([])  # field_descriptor
+        self.method_names = set([])  # package_name_method_name
```
Comment (harupy) [inline mlflow/protos/proto_to_graphql/code_generator.py:21]:
```suggestion
        self.queries = set()  # method_descriptor
        self.mutations = set()  # method_descriptor
        self.inputs = []  # field_descriptor
        self.types = []  # field_descriptor
        self.enums = set()  # enum_descriptor
        self.outputs = set()  # field_descriptor
        self.method_names = set()  # package_name_method_name
```

File: mlflow/protos/proto_to_graphql/autogeneration_utils.py
Comment (harupy) [inline mlflow/protos/proto_to_graphql/autogeneration_utils.py:1]: Can we create a `dev/proto_to_graphql` directory and move the files there?

Comment (edwardfeng-db) [reply mlflow/protos/proto_to_graphql/autogeneration_utils.py:1]: Yeah make sense, sounds good

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**